src/App.tsx

```diff
@@ -44,6 +44,16 @@ function App() {
     textarea
   );
   const setTextToParseThrottle = useThrottleCallback(setTextToParse, 2);
+  const inputRef = useRef({} as HTMLTextAreaElement);
+  const [currentPosition, setCurrentPosition] = useState({
```

Comment:
Because we're using `useState` here, it will trigger a re-render of the component when the cursor position changes, but actually we just want to set the cursor position after the text has changed, without re-rending the component. So it might be better to use `useRef` instead